### PR TITLE
Add ability to delete attachment file and resource

### PR DESCRIPTION
**Pull request by rsachdevch**

### CODE REVIEW

File: handlers/attachment_resource.go [hunk 1]
```diff
@@ -159,3 +159,50 @@ func HandleGetAttachmentDetails(svc dao.Service) http.Handler {
 		utils.WriteJSONWithStatus(w, req, attachmentResponse, http.StatusOK)
 	})
 }
+
+func HandleDeleteAttachment(svc dao.Service) http.Handler {
```
Comment (emead) [inline handlers/attachment_resource.go:163]: Exported function should have a comment

File: dao/mongo.go [hunk 2]
```diff
@@ -431,24 +431,55 @@ func (m *MongoService) GetAttachmentFromInsolvencyResource(transactionID string,
...
+	// Choose specific transaction for insolvency case with attachment to be removed
+	filter := bson.M{"transaction_id": transactionID}
+
+	// Check if insolvency case exists for specified transactionID
+	storedInsolvency := collection.FindOne(context.Background(), filter)
+	err := storedInsolvency.Err()
+	if err != nil {
+		if err == mongo.ErrNoDocuments {
+			log.Error(err)
+			return http.StatusNotFound, fmt.Errorf("there was a problem handling your request for transaction id [%s] - insolvency case not found", transactionID)
```
Comment (emead) [inline dao/mongo.go:462]: Shouldn't return an error here. http.StatusNotFound is enough to indicate that no document was found

File: handlers/attachment_resource.go [hunk 3]
```diff
@@ -159,3 +159,50 @@ func HandleGetAttachmentDetails(svc dao.Service) http.Handler {
...
+			status, err := utils.ResponseTypeToStatus(responseType.String())
+			if err != nil {
+				w.WriteHeader(http.StatusInternalServerError)
+				return
+			}
+			w.WriteHeader(status)
+			return
+		}
+
+		// Delete attachment from Mongo
```
Comment (emead) [inline handlers/attachment_resource.go:197]: The handler doesn't know which DB is used
```suggestion
		// Delete attachment from DB
```

File: handlers/register.go [hunk 4]
```diff
@@ -39,6 +39,7 @@ func Register(mainRouter *mux.Router, svc dao.Service) {
 
 	appRouter.Handle("/{transaction_id}/insolvency/attachments", HandleSubmitAttachment(svc)).Methods(http.MethodPost).Name("submitAttachment")
 	appRouter.Handle("/{transaction_id}/insolvency/attachments/{attachment_id}", HandleGetAttachmentDetails(svc)).Methods(http.MethodGet).Name("GetAttachmentDetails")
+	appRouter.Handle("/{transaction_id}/insolvency/attachments/{attachment_id}", HandleDeleteAttachment(svc)).Methods(http.MethodDelete).Name("DeleteAttachment")
```
Comment (emead) [inline handlers/register.go:42]: Please add unit test to register_test.go

File: dao/mongo.go (hunk 2 above)
Comment (rsachdevch) [reply dao/mongo.go:462]: The error is used in the message response for the user to distinguish from other error messages in that method. It's also consistent with what we've done in other parts of our app

**Review by Joseph-Nash — APPROVED** [review]

**Pull request merged**